experiments/main.py

```diff
@@ -0,0 +1,410 @@
+
+    checkpoint = None
+    non_loadable_attributes = ["logging_path", "dataset_path", "batch_size"]
+    temp = {key: val for key, val in config.items() if key in non_loadable_attributes}
+    checkpoint_path = os.path.join(config['logging_path'], 'latest_model')
+    json_logs_file_name = 'jsonlogs.jsonl'
+    if os.path.isfile(checkpoint_path):
+        logging.basicConfig(filename=os.path.join(config['logging_path'], "logs.txt"),
+                            filemode='a+',
+                            format='%(asctime)s,%(msecs)d %(name)s %(levelname)s %(message)s',
+                            datefmt='%H:%M:%S',
+                            level=logging.INFO)
+        if distributed:
+            print_msg(f"\n\nLoading checkpoint {checkpoint_path} on gpu {dist_args['rank']}")
+            checkpoint = torch.load(checkpoint_path, map_location=f"cuda:{dist_args['gpu']}")
+        else:
+            print_msg(f"\n\nLoading checkpoint {checkpoint_path}")
+            checkpoint = torch.load(checkpoint_path)
+        config = checkpoint['config']
+        for key in non_loadable_attributes:
+            config[key] = temp[key]
+        if distributed:
+            print_msg(f"Loaded the checkpoint successfully on gpu {dist_args['rank']}")
+        else:
+            print_msg(f"Loaded the checkpoint successfully")
+
+        if rank == 0:
+            print_msg(f"Resuming from task {config['cur_task_id']} epoch {config['task_epoch']}")
+            # Remove logs related to traing after the checkpoint was saved
+            utils.remove_extra_logs(config['cur_task_id'], config['task_epoch'],
+                                    os.path.join(config['logging_path'], json_logs_file_name))
+            if distributed:
+                dist.barrier()
+        else:
+            dist.barrier()
+    else:
+        if rank == 0:
+            os.makedirs(config['logging_path'], exist_ok=True)
+            if os.path.isfile(os.path.join(config['logging_path'], json_logs_file_name)):
+                os.remove(os.path.join(config['logging_path'], json_logs_file_name))
+            logging.basicConfig(filename=os.path.join(config['logging_path'], "logs.txt"),
+                                filemode='w',
+                                format='%(asctime)s,%(msecs)d %(name)s %(levelname)s %(message)s',
+                                datefmt='%H:%M:%S',
+                                level=logging.INFO)
+            if distributed:
+                dist.barrier()
+        else:
+            dist.barrier()
+            logging.basicConfig(filename=os.path.join(config['logging_path'], "logs.txt"),
+                                filemode='a',
+                                format='%(asctime)s,%(msecs)d %(name)s %(levelname)s %(message)s',
+                                datefmt='%H:%M:%S',
+                                level=logging.INFO)
+
+    torch.random.manual_seed(config['seed'])
+    np.random.seed(config['seed'])
```

Comment:
If you are generating random numbers with numpy in your workers, they will all generate the same random numbers. If this is what you want, that is fine, if not, then be very careful with these things. E.g., if you are using workers to randomly sample from your dataset, all workers will fetch the same samples.